### PR TITLE
Small docs changes based on a conversation with Dave

### PR DESCRIPTION
**Pull request by wxtim**
A couple of changes based on a conversation with @dpmatthews

### CODE REVIEW

File: sphinx/api/configuration/suite.rst [hunk 1]
```diff
@@ -47,7 +47,7 @@ A suite directory may contain the following:
 
    .. rose:conf:: env
 
-      Specify the environment variables to export to the suite daemon. The
+      Specify the environment variables to export to the workflow scheduler. The
```
Comment (oliver-sanders) [inline sphinx/api/configuration/suite.rst:50]: Can make this a `:term:'scheduler'`

File: sphinx/api/environment-variables.rst [hunk 2]
```diff
@@ -225,14 +225,17 @@ Rose Environment Variables
...
-      The value of :rose:conf:`rose.conf|site` setting.
-   Provided By
-      * :ref:`Cylc Rose` and :ref:`cylc install <Install-Workflow>`.
+   .. deprecated:: 2.0.0
+
+      This variable is no longer available.
+
+      .. note::
+
+         :ref:`rose-stem` provides this variable.
```
Comment (oliver-sanders) [inline sphinx/api/environment-variables.rst:238]: This is a bit confusing, is it deprecated or supported? It's actually obsolete.

Perhaps explain that setting it "globally" is no longer supported, however, via `rose-stem` is still supported.

File: sphinx/tutorial/rose/suites.rst [hunk 3]
```diff
@@ -48,7 +48,7 @@ A Rose suite configuration is a Cylc :term:`source directory` containing a
    :rose:file:`rose-app.conf` file. Its main configuration sections are:
 
    :rose:conf:`rose-suite.conf[env]`
-      Environment variables for use by the whole suite.
+      Environment variables for use by the workflow scheduler.
```
Comment (oliver-sanders) [inline sphinx/tutorial/rose/suites.rst:51]: Can link to the Cylc glossary.

File: sphinx/api/configuration/suite.rst (hunk 1 above)
Comment (wxtim) [reply sphinx/api/configuration/suite.rst:50]:
```suggestion
       Specify the environment variables to export to the workflow :term:`scheduler`. The

File: sphinx/tutorial/rose/suites.rst (hunk 3 above)
Comment (wxtim) [reply sphinx/tutorial/rose/suites.rst:51]:
```suggestion
       Environment variables for use by the workflow :term:`scheduler`. 
```

**Review by oliver-sanders — APPROVED** [review]

**Pull request merged**